homeassistant/components/homekit_controller/__init__.py

```diff
@@ -48,7 +48,7 @@ def __init__(self, accessory: HKDevice, devinfo):
         self._features = 0
         self.setup()
 
-        self._signals = []
+        self._signals: list[Any] = []
```

Comment:
We could replace the need for this attribute by using `self.async_on_remove` to handle the unsubscribe callbacks of the dispatcher. That's the preferred way nowadays.

Reply:
Thats a good tip! Applied, and tests still passing locally. Cheers.